examples/leasing/node_release_script.py

```diff
@@ -0,0 +1,241 @@
+
+
+def update_project_in_status_file(
+        statusfile, node, new_project, old_project
+        ):
+    """Project will be changed in
+    file to match the project in HIL"""
+
+    newline = ''
+    with open(statusfile, 'r') as status_file,\
+            open('/tmp/tempfile', 'w') as output_file:
+        for line in status_file:
+            words = line.split()
+            if node == words[0]:
+                words[1] = new_project
+                newline = ' '.join(words) + "\n"
+                output_file.write(newline)
+            else:
+                output_file.write(line)
+    status_file.close()
+    output_file.close()
+    remove(statusfile)
+    move('/tmp/tempfile', statusfile)
+
+
+def detach_networks_from_node(hil_client, node):
+    """Disconnect all networks from all of the node's NICs
+    get node information and then iterate on the nics"""
+
+    node_info = hil_client.node.show(node)
+
+    for nic in node_info['nics']:
+        # get the port and switch to which the nics are connected to
+        port = nic['port']
+        switch = nic['switch']
+        if port and switch:
+            try:
+                hil_client.port.port_revert(switch, port)
+                print('Removed all networks from node `%s`' % node)
+            except FailedAPICallException:
+                print('Failed to revert port `%s` on node \
+                        `%s` switch `%s`' % (port, node, switch))
+                raise HILClientFailure()
+
+
+def release_from_project(
+        hil_client, statusfile, node, project
+        ):
+    """Updates the file with project name as free and
+    releases it back to free pool"""
+
+    time.sleep(2)
+    # tries 2 times to detach the project because there might be a pending
+    # networking action setup by revert port in the previous step.
+    counter = 2
+    while counter:
+
+        try:
+            hil_client.project.detach(project, node)
+            print('Node `%s` removed from project `%s`' % (node, project))
+            break
+        except FailedAPICallException as ex:
+            if ex.message == 'Node has pending network actions':
+                counter -= 1
+                time.sleep(2)
+            else:
+                print('HIL reservation failure: Unable to \
+                        detach node `%s` from project `%s`' % (node, project))
+                raise HILClientFailure(ex.message)
+    if counter == 0:
+        print('HIL reservation failure: Unable to detach node \
+                `%s` from project `%s`' % (node, project))
+        raise HILClientFailure()
+
+    with open(statusfile, 'r') as status_file, \
+            open('/tmp/tempfile', 'w') as output_file:
+        for line in status_file:
+            words = line.split()
+            if node == words[0]:
+                newline = node + " " + "free_pool " + str(0) + "\n"
+                output_file.write(newline)
+            else:
+                output_file.write(line)
```

Comment:
Rather than having the else clause, you could just over-ride the existing variable:

```python
if node == words[0]:
    line = node + " free_pool 0\n"
output_file.write(line)
```